rfcs/rfc0009.md

```diff
@@ -0,0 +1,213 @@
+
+* Most of the `Scalar::Util` functions should be candidates
+
+* Most of the `List::Util` functions that do not act as higher-order functionals can probably be included. This would be functions like `sum`, `max`, `pairs`, `uniq`, `head`, etc. The higher-order functionals such as `reduce` or its specialisations like `first` and `any` would not be candidates, because of their "block-like function as first argument" parsing behaviour at compiletime.
+
+* Some of the `POSIX` functions that act abstractly as in-memory data utilities, such as `ceil` and `floor`. I would not recommend adding the bulk of the operating system interaction functions from POSIX.
+
+* There are other RFCs or Pre-RFC discussions that suggest adding new named functions that would be good candidates for this module. They can be considered on their own merit, by reference to this RFC. At time of writing this may include new functions to handle core-supported boolean types (RFC 0008) or the new module-loading function (RFC 0006).
+
+* Once a stable set of functions is defined, consider creating version-numbered bundles in a similar theme to those provided by `feature.pm`:
+
+```perl
+use builtin ':5.40';  # imports all those functions defined by perl v5.40
+```
+
+* Once version-numbered bundles exist, consider whether the main `use VERSION` syntax should also enable them; i.e.
+
+```perl
+use v5.40;  # Does this imply  use builtin ':5.40'; ?
+```
+
+## Rejected Ideas
+
+### Multiple Namespaces
+
+An initial discussion had been to consider giving multiple namespaces for these functions to live in, such as `string::` or `ref::`. That was eventually rejected as being overly complex, and inviting a huge number of new functions. By sticking to a single namespace for all regular functions, we apply a certain amount of constraining pressure to limit the number of such functions that are provided.
+
+## Open Issues
+
+### Package Name
+
+What is the package name these functions are provided in?
+
+The discussion above used `builtin::`. Other proposed suggestions include `function::` or `std::`.
+
+### Pragma Module Name
+
+What is the module name for the lexical-import pragma?
+
+The discussion above used `use builtin`, to match the package name. Technically it does not have to match the package name. In particular, if during implementation or initial use it is found to be problematic that the name does match, the import module could use a plural form of the same word; as
+
+```perl
+use builtins qw( function names here );
+```
+
+### Version Numbering
+
+How to version number the `builtin` pragma module?
+
+This becomes an especially interesting when considering the dual-life module distribution provided as a polyfill for older perls.
+
+A case can be made that its version number should match the version of perl itself for which it provides polyfill functions. Thus, code could write:
+
+```perl
+use builtin v5.40;
+# and now all the builtin:: functions from perl v5.40 are available
```

Comment:
In addition to the bugfix concerns of tying the module version to the Perl version, this shouldn't import any functions without either a list of functions or the aforementioned bundle tags. The version would instead be useful to create a nicer error message if your version of builtin.pm is not new enough:

```perl
use builtin v1.2.3 'fribble';
# after ascertaining that this is the version that first provides the fribble function
```

So the module could have its own version scheme like any other module, and additionally document both the module version and Perl version wherein each function becomes available. I don't think this will be too confusing as long as the versions are not similar to each other, as the module version would be singularly useful for the specific mechanism above.